lib/ParseSIL/ParseSIL.cpp

```diff
@@ -3117,6 +3117,29 @@ bool SILParser::parseSILInstruction(SILBuilder &B) {
   }
   // SWIFT_ENABLE_TENSORFLOW
   case SILInstructionKind::GraphOperationInst: {
+    bool noClustering = false;
+    if (P.consumeIf(tok::l_square)) {
+      Identifier ident;
+      SourceLoc identLoc;
+      if (parseSILIdentifier(ident, identLoc,
+                             diag::expected_in_attribute_list)) {
+        P.diagnose(P.Tok, diag::expected_tok_in_sil_instr,
+                   "'no_clustering' attribute");
+        return true;
+      }
+      StringRef attr = ident.str();
+
+      if (attr == "no_clustering")
```

Comment:
When `[` is entered, the parser should expect to see `no_clustering`, otherwise it would error out. So `noClustering = true;` can be set earlier once you enter the condition on line 3121.

No need for a pair of if-else.

```cpp
if (attr != "no_clustering") {
  P.diagnose(...);
  return true;
}
```

Reply:
Good point. This simplified the code quite a bit. Thanks!